huaweicloud/services/secmaster/data_source_huaweicloud_secmaster_workspaces.go

```diff
@@ -0,0 +1,222 @@
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The workspace name.`,
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The workspace description.`,
+						},
+						"enterprise_project_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The enterprise project ID.`,
+						},
+						"enterprise_project_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The enterprise project name.`,
+						},
+						"is_view": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: `Whether the view is used.`,
+						},
+						"view_bind_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The space ID bound to the view.`,
+						},
+						"view_bind_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The space name bound to the view.`,
+						},
+						"created_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The creation time.`,
+						},
+						"updated_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The update time.`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+type WorkspacesDSWrapper struct {
+	*schemas.ResourceDataWrapper
+	Config *config.Config
+}
+
+func newWorkspacesDSWrapper(d *schema.ResourceData, meta interface{}) *WorkspacesDSWrapper {
+	return &WorkspacesDSWrapper{
+		ResourceDataWrapper: schemas.NewSchemaWrapper(d),
+		Config:              meta.(*config.Config),
+	}
+}
+
+func dataSourceSecmasterWorkspacesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	wrapper := newWorkspacesDSWrapper(d, meta)
+	listWorkspacesRst, err := wrapper.ListWorkspaces()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(id)
+
+	err = wrapper.listWorkspacesToSchema(listWorkspacesRst)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+// @API SecMaster GET /v1/{project_id}/workspaces
+func (w *WorkspacesDSWrapper) ListWorkspaces() (*gjson.Result, error) {
+	client, err := w.NewClient(w.Config, "secmaster")
+	if err != nil {
+		return nil, err
+	}
+
+	uri := "/v1/{project_id}/workspaces"
+	params := map[string]any{
+		"name":                  w.Get("name"),
+		"description":           w.Get("description"),
+		"view_bind_id":          w.Get("view_bind_id"),
+		"view_bind_name":        w.Get("view_bind_name"),
+		"create_time_start":     w.Get("create_time_start"),
+		"create_time_end":       w.Get("create_time_end"),
+		"is_view":               w.Get("is_view", true),
+		"ids":                   w.Get("ids"),
+		"enterprise_project_id": w.Get("enterprise_project_id"),
```

Comment:
the enterprise_project_id should support provider-level config,
better to use config.DataGetEnterpriseProjectID